partiql-lang/src/main/kotlin/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt

```diff
@@ -368,37 +369,23 @@ internal class AstToLogicalVisitorTransform(
                     }
                 }
 
-                val target = dmlOp.target.toDmlTargetId()
+                val target = dmlOp.target.toDmlTarget()
                 val alias = dmlOp.asAlias?.let {
                     PartiqlLogical.VarDecl(it)
-                } ?: PartiqlLogical.VarDecl(target.name)
-
-                val operation = when (val conflictAction = dmlOp.conflictAction) {
-                    null -> PartiqlLogical.DmlOperation.DmlInsert(targetAlias = alias)
-                    is PartiqlAst.ConflictAction.DoReplace -> when (conflictAction.value) {
-                        is PartiqlAst.OnConflictValue.Excluded -> PartiqlLogical.DmlOperation.DmlReplace(
-                            targetAlias = alias,
-                            condition = conflictAction.condition?.let { transformExpr(it) },
-                            rowAlias = conflictAction.condition?.let { PartiqlLogical.VarDecl(SymbolPrimitive(EXCLUDED, emptyMetaContainer())) }
-                        )
-                    }
-                    is PartiqlAst.ConflictAction.DoUpdate -> when (conflictAction.value) {
-                        is PartiqlAst.OnConflictValue.Excluded -> PartiqlLogical.DmlOperation.DmlUpdate(
-                            targetAlias = alias,
-                            condition = conflictAction.condition?.let { transformExpr(it) },
-                            rowAlias = conflictAction.condition?.let { PartiqlLogical.VarDecl(SymbolPrimitive(EXCLUDED, emptyMetaContainer())) }
-                        )
-                    }
-                    is PartiqlAst.ConflictAction.DoNothing -> TODO("`ON CONFLICT DO NOTHING` is not supported in logical plan yet.")
-                }
+                } ?: PartiqlLogical.VarDecl(target.identifier.name)
 
-                PartiqlLogical.Statement.Dml(
-                    target = target,
-                    operation = operation,
-                    rows = transformExpr(dmlOp.values),
-                    metas = node.metas
-                )
+                PartiqlLogical.build {
+                    dmlInsert(
+                        target = target,
+                        targetAlias = alias,
+                        rowsToInsert = transformExpr(dmlOp.values),
+                        metas = node.metas,
+                        onConflict = transformConflictAction(dmlOp.conflictAction)
+                    )
+                }
             }
+            // DL TODO: transformDmlUpdate
```

Comment:
DL TODO, does this still need to be fleshed out here?

Reply:
Looks like this is handled by the DmlOp.Set branch below if I'm reading this right, I think this comment can just be removed